src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -92,35 +108,59 @@ private void writeHeader(FileChannel fc, AtomicLong pointer, int version, long f
      * @param address The address of the entry.
      * @return The log unit entry at that address, or NULL if there was no entry.
      */
-    private LogData readEntry(FileHandle fh, long address)
+    private LogData readRecord(FileHandle fh, long address)
             throws IOException {
-        ByteBuffer o = fh.getMapForRegion(64, (int) fh.getChannel().size());
+        ByteBuffer o = fh.getMapForRegion(LogFileHeader.size, (int) fh.getChannel().size());
         while (o.hasRemaining()) {
-            short magic = o.getShort();
-            if (magic != 0x4C45) {
+            short delimiter = o.getShort();
+            if (delimiter != recordDelimiter) {
                 return null;
             }
-            short flags = o.getShort();
-            long addr = o.getLong();
+
+            Checksum checksumRead = new BufferChecksum();
+            checksumRead.deserialize(o);
+
+            ByteBuffer checksumBuf = o.slice();
+            long recordAddress = o.getLong();
+            int dataSize = o.getInt();
+            int metaDataSize = o.getInt();
+            int entrySize = dataSize + metaDataSize;
+
+            int recordLen = 8 // size of address
+                          + 4 // size of int for dataSize
+                          + 4 // size of int for metaDataSize
+                          + entrySize;
+
+            if(checksum) {
+                // Compute record checksum
+                byte[] bytes = new byte[recordLen];
+                checksumBuf.limit(recordLen);
+                checksumBuf.get(bytes);
+
+                Checksum recomputedChecksum = getChecksum(bytes);
+                if(!checksumRead.equals(recomputedChecksum)) {
+                    log.error("Checksum mismatch possibly around address {}, Stream log file {}", address,
```

Comment:
Indeed it is a mismatch, meant to "Checksum mismatch, possibly around address ". Since the address is also part of the checksum value, then if there is a checksum mismatch, you can't tell if the address or the logdata are corrupted. Bits in the address field can flip, so we can't definately say that the log entry at address X is corrupted.